package/system/zram-swap/files/zram.init

```diff
@@ -163,16 +163,15 @@ start()
 	local zram_size="$( zram_getsize )"
 	local zram_dev="$( zram_getdev )"
 	zram_applicable "$zram_dev" || return 1
-	local zram_priority="$( uci -q get system.@system[0].zram_priority )"
-	zram_priority=${zram_priority:+-p $zram_priority}
+	local zram_priority=`cat /proc/swaps | tail -n +2 | awk '{print $5}' | sort | tail -n 1 | { grep . || echo -1; }`
 
 	logger -s -t zram_start -p daemon.debug "activating '$zram_dev' for swapping ($zram_size MegaBytes)"
 
 	zram_reset "$zram_dev" "enforcing defaults"
 	zram_comp_algo "$zram_dev"
 	echo $(( $zram_size * 1024 * 1024 )) >"/sys/block/$( basename "$zram_dev" )/disksize"
 	/sbin/mkswap "$zram_dev"
-	/sbin/swapon -d $zram_priority "$zram_dev"
+	/sbin/swapon -d -p "$(( $zram_priority < 0 ? 0 : $zram_priority+1 ))" "$zram_dev"
```

Comment:
Typically, no $ is needed inside (()) brackets.

Reply:
> 
> 
> Typically, no $ is needed inside (()) brackets.

I removed the double quotes, but `$` seems to be required though

>  zram-swap requires PKG_RELEASE bump
It set to [autorelease](https://github.com/openwrt/openwrt/blob/88114f617ae7bffe13d19d7b9575659a3d3cd9b6/package/system/zram-swap/Makefile#L8), not sure if any change is required there.

> commit message seems to contain lines with more than 74 chars.
Yeah , I reduced, thanks

Reply:
> > Typically, no $ is needed inside (()) brackets.
> 
> I removed the double quotes, but `$` seems to be required though
You mean $ inside the double parentheses or outside? Outside you need one, yes, but not inside. Try this on your OpenWrt host:
`root@OpenWrt:~# zram_priority=50`
`root@OpenWrt:~# echo $(( zram_priority < 0 ? 0 : zram_priority+1 ))`
`51`
Whether you use double quotes around your arithmetic statement or not, doesn't really matter in this case and it works as well with other initial values like -1.

Aside from that. I just realize, you are basically deprecating the uci option zram_priority with your patch here.

Reply:
@silentcreek  That option was added in [this commit](https://patchwork.ozlabs.org/project/openwrt/patch/20200109182744.32700-1-m.storchak@gmail.com/) , it's main purpose is to assign higher priority to zram than other swaps, as my patch is already  assigning higher priority to zram than other swaps, I am not sure if it is still required.